Makefile

```diff
@@ -6,16 +6,45 @@ SHELL = /bin/sh
 export DOCKER_IMAGE_NAME ?= jupyter-math
 export DOCKER_IMAGE_TAG ?= 2.0.5
 
-.PHONY: publish-local-registry
-publish-local-registry:
-	docker tag simcore/services/dynamic/${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG} registry:5000/simcore/services/dynamic/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)
-	docker push registry:5000/simcore/services/dynamic/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)
-	@curl registry:5000/v2/_catalog | jq
+
+define _bumpversion
+	# upgrades as $(subst $(1),,$@) version, commits and tags
+	@docker run -it --rm -v $(PWD):/${DOCKER_IMAGE_NAME} \
+		-u $(shell id -u):$(shell id -g) \
+		itisfoundation/ci-service-integration-library:v1.0.1-dev-25 \
+		sh -c "cd /${DOCKER_IMAGE_NAME} && bump2version --verbose --list --config-file $(1) $(subst $(2),,$@)"
+endef
+
+.PHONY: version-patch version-minor version-major
+version-patch version-minor version-major: .bumpversion.cfg ## increases service's version
+	@make compose-spec
+	@$(call _bumpversion,$<,version-)
+	@make compose-spec
+
+.PHONY: compose-spec
+compose-spec: ## runs ooil to assemble the docker-compose.yml file
+	@docker run -it --rm -v $(PWD):/${DOCKER_IMAGE_NAME} \
+		-u $(shell id -u):$(shell id -g) \
+		itisfoundation/ci-service-integration-library:v1.0.1-dev-25 \
+		sh -c "cd /${DOCKER_IMAGE_NAME} && ooil compose"
 
 .PHONY: build
-build:
+build: compose-spec	## build docker image
 	docker-compose build
 
 .PHONY: run-local
-run-local:
-	docker-compose --file docker-compose-local.yml up
\ No newline at end of file
+run-local:	## runs image with local configuration
+	docker-compose --file docker-compose-local.yml up
+
+.PHONY: publish-local
+publish-local:
+	docker tag simcore/services/dynamic/${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG} registry:5000/simcore/services/dynamic/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)
```

Comment:
I think the docker-compose produced by ooil tags (``ooil compose``) tags the service with prefix ``registry:500`` ... 

See for instance  in https://github.com/pcrespov/osparc-function-services/blob/is353/en-sensitivity-ua/docker-compose.yml#L22

Reply:
In this case, the following compose spec is generated
 I don't remember precisely but I might have changed the default behaviour to not use `registry:5000`.


```yaml
version: '3.7'
services:
  jupyter-math:
    build:
      context: ./
      dockerfile: Dockerfile
      args:
        VERSION: 2.0.5
      labels:
        org.label-schema.build-date: '2022-04-29T07:49:20Z'
        org.label-schema.schema-version: '1.0'
        org.label-schema.vcs-ref: 741250d
        org.label-schema.vcs-url: git@github.com:ITISFoundation/jupyter-math.git
        io.simcore.name: '{"name": "JupyterLab Math (Python+Octave)"}'
        io.simcore.thumbnail: '{"thumbnail": "https://upload.wikimedia.org/wikipedia/commons/thumb/3/38/Jupyter_logo.svg/1024px-Jupyter_logo.svg.png"}'
        io.simcore.description: '{"description": "[JupyterLab](https://jupyter.org/)
          with **math** kernels in [octave](https://www.gnu.org/software/octave/index)
          (largely compatible with Matlab code) and python. Python kernel includes
          popular math packages like [numpy](https://numpy.org/), [pandas](https://pandas.pydata.org/),
          [matplotlib](https://matplotlib.org/), [plotly](https://plotly.com/python/)
          and [scipy](https://www.scipy.org/) packages (see requirements.txt list)\n"}'
        io.simcore.key: '{"key": "simcore/services/dynamic/jupyter-math"}'
        io.simcore.version: '{"version": "2.0.5"}'
        io.simcore.integration-version: '{"integration-version": "2.0.0"}'
        io.simcore.type: '{"type": "dynamic"}'
        io.simcore.authors: '{"authors": [{"name": "Pedro Crespo-Valero", "email":
          "crespo@itis.swiss", "affiliation": "IT''IS Foundation"}, {"name": "Sylvain
          Andereregg", "email": "anderegg@itis.swiss", "affiliation": "IT''IS Foundation"},
          {"name": "Sylvain Reboux", "email": "reboux@zmt.swiss", "affiliation": "Zurich
          MedTech"}]}'
        io.simcore.contact: '{"contact": "crespo@itis.swiss"}'
        io.simcore.inputs: '{"inputs": {"input_1": {"displayOrder": 1.0, "label":
          "input_files_1", "description": "Any input files. One or several files compressed
          in a zip will be downloaded in an inputs folder.", "type": "data:*/*"},
          "input_2": {"displayOrder": 2.0, "label": "input_files_2", "description":
          "Any input files. One or several files compressed in a zip will be downloaded
          in an inputs folder.", "type": "data:*/*"}, "input_3": {"displayOrder":
          3.0, "label": "input_files_3", "description": "Any input files. One or several
          files compressed in a zip will be downloaded in an inputs folder.", "type":
          "data:*/*"}, "input_4": {"displayOrder": 4.0, "label": "input_files_4",
          "description": "Any input files. One or several files compressed in a zip
          will be downloaded in an inputs folder.", "type": "data:*/*"}}}'
        io.simcore.outputs: '{"outputs": {"output_1": {"displayOrder": 1.0, "label":
          "Output files 1", "description": "Output files uploaded from the outputs
          folder", "type": "data:*/*"}, "output_2": {"displayOrder": 2.0, "label":
          "Output files 2", "description": "Output files uploaded from the outputs
          folder", "type": "data:*/*"}, "output_3": {"displayOrder": 3.0, "label":
          "Output files 3", "description": "Output files uploaded from the outputs
          folder", "type": "data:*/*"}, "output_4": {"displayOrder": 4.0, "label":
          "Output files 4", "description": "Output files uploaded from the outputs
          folder", "type": "data:*/*"}}}'
        io.simcore.boot-options: '{"boot-options": {"boot_mode": {"label": "Boot mode",
          "description": "Select boot type for the service", "default": "0", "items":
          {"0": {"label": "JupyterLab", "description": "Display the JupyterLab interface
          the default boot mode"}, "1": {"label": "Voila", "description": "To start
          as Voila save a notebook as \"voila.ipynb\" in the root folder"}}}}}'
        simcore.service.restart-policy: no-restart
        simcore.service.paths-mapping: '{"inputs_path": "/home/jovyan/work/inputs",
          "outputs_path": "/home/jovyan/work/outputs", "state_paths": ["/home/jovyan/work/workspace"]}'
        simcore.service.settings: '[{"name": "ports", "type": "int", "value": 8888},
          {"name": "constraints", "type": "string", "value": ["node.platform.os ==
          linux"]}, {"name": "Resources", "type": "Resources", "value": {"Limits":
          {"NanoCPUs": 4000000000, "MemoryBytes": 17179869184}}}]'
    image: simcore/services/dynamic/jupyter-math:2.0.5
```